### PR TITLE
Search results title

### LINKED ISSUE #576: Search results title
The newly-redesigned search results pages show the search query as a title, above the results. This title is a quite a large size on the 'all' results view, and smaller on other endpoints.

All results figma: https://www.figma.com/file/w60dl1XPUvSaRncv1Utmnb/Openverse-Releases?node-id=133%3A22543
Other results figma: https://www.figma.com/file/w60dl1XPUvSaRncv1Utmnb/Openverse-Releases?node-id=133%3A14189

### PR DESCRIPTION
**Pull request by zackkrida**
## Fixes

Fixes #576 by @zackkrida

## Description

Adds a super simple component to display the search term as an `<h1>` title on search result pages. Accepts a size prop to support the larger size used on the 'all' results page. Hidden on mobile but still accessible by screen readers.

Spacing for this component will be handled by it's parent; currently in the 'transitional' view the horizontal and vertical padding are incorrect. The line height however, is intentionally correct per the mockup.

CleanShot 2022-01-05 at 16 27 00@2x
CleanShot 2022-01-05 at 16 27 04@2x

## Testing Instructions

View the 'all' and 'images' tabs, observe the visible title and size difference. Note that this component doesn't display on unsupported media types like video. 

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by panchovm** [comment]
In **All** tab looks great, but I noticed that is bigger (`57px`) in **Image** and **Audio** tabs. It should be `32px` as **Heading 1** Figma style.

I also noticed that I did not add the biggest heading size in the Design Library. I will do it now.

**Review by panchovm — CHANGES_REQUESTED** [review]
Text style correction as pointed in the comment.

**Comment by zackkrida** [comment]
@panchovm good find, I will fix the smaller size—thank you!

**Comment by zackkrida** [comment]
@panchovm Fixed!

CleanShot 2022-01-06 at 10 56 49@2x

File: test/unit/specs/components/v-search-results-title.spec.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import VSearchResultsTitle from '~/components/VSearchResultsTitle'
+import { render, screen } from '@testing-library/vue'
+
+describe('VSearchResultsTitle', () => {
+  let options = {
+    props: {
+      size: 'large',
+    },
+    scopedSlots: {
```
Comment (obulat) [inline test/unit/specs/components/v-search-results-title.spec.js:9]: Why is it `scopedSlots`, not simple `slots` here?

**Review by obulat — APPROVED** [review]
LGTM!

File: test/unit/specs/components/v-search-results-title.spec.js (hunk 1 above)
Comment (zackkrida) [reply test/unit/specs/components/v-search-results-title.spec.js:9]: I copy/pasted from another test, I wonder if it makes a difference? I can switch to `slots`.

Comment (zackkrida) [inline test/unit/specs/components/v-search-results-title.spec.js:9]:
```suggestion
    slots: {
```

File: src/components/VSearchResultsTitle.vue [hunk 2]
```diff
@@ -0,0 +1,27 @@
+<template>
+  <h1
+    class="capitalize leading-tight sr-only md:not-sr-only"
```
Comment (zackkrida) [inline src/components/VSearchResultsTitle.vue:3]:
```suggestion
    class="uppercase leading-tight sr-only md:not-sr-only"
```

This is probably preferable.

**Review by panchovm — APPROVED** [review]
It looks great to me. Adding here what I chatted with @zackkrida.

Initially, the biggest size had a letter spacing of -3px, but we decided to remove it since I'm uncertain about its impact on i18n and other alphabets. On the other hand, the H1 text capitalized the first letter of each word, and we decided to capitalize the first word.

**Pull request merged**